frontend/src/metabase/query_builder/selectors.js

```diff
@@ -561,11 +561,13 @@ export const getIsDirty = createSelector(
 export const getIsSavedQuestionChanged = createSelector(
   [getQuestion, getOriginalQuestion],
   (question, originalQuestion) => {
+    const isSavedQuestion = originalQuestion != null;
+    const hasChanges = question != null;
+    const wereChangesSaved = question?.isSaved();
+    const hasUnsavedChanges = hasChanges && !wereChangesSaved;
```

Comment:
I could not understand why this function is called `getIsSavedQuestionChanged` yet it checks for `!question.isSaved()`. 

Kelvin explained it to me, and then I introduced some explanatory variables here to help it make more clear what's going on.

Reply:
Thanks. This looks more readable now.